### PR TITLE
Remove delete db button from db list

### LINKED ISSUE #19869: Remove "Delete" button from Admin > Databases overview
I've seen way too many delete their entire database. Deleting a database should be as difficult as possible.

Let's see if that solves the problem. Otherwise we night need to make it a lot more explicit that deleting a database will **delete** all questions etc connected to this database, but there's #11381 for that.

### PR DESCRIPTION
**Pull request by gusaiani**
Resolves #19869

### How to Test

1. Go to http://localhost:3000/admin/databases
2. Hover over database items

You should no longer see a delete button for any db.

### CODE REVIEW

File: frontend/test/metabase/scenarios/admin/databases/list.cy.spec.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ describe("scenarios > admin > databases > list", () => {
 
   it("should not let you see saved questions in the database list", () => {
```
Comment (gusaiani) [inline frontend/test/metabase/scenarios/admin/databases/list.cy.spec.js:16]: @nemanjaglumac Does this test need to exist? Imo it does not.

**Comment by nemanjaglumac** [comment]
So, will users now be able to delete database if they want to? We didn't provide any alternative?

File: frontend/test/metabase/scenarios/admin/databases/list.cy.spec.js (hunk 1 above)
Comment (nemanjaglumac) [reply frontend/test/metabase/scenarios/admin/databases/list.cy.spec.js:16]: I guess there was some weird bug back then when that test was created? o.O

**Review by flamber — APPROVED** [review]
:+1: 
I delete databases every day, I don't mind the extra click, and hopefully it will prevent some accidental trigger-finger deletions. In a perfect work the deletion dialog would be a lot more explicit about what will happen and a count of questions/etc that will be permanently deleted.

File: frontend/test/metabase/scenarios/admin/databases/list.cy.spec.js [hunk 2]
```diff
@@ -37,32 +37,6 @@ describe("scenarios > admin > databases > list", () => {
     cy.url().should("match", /\/admin\/databases\/1$/);
   });
 
-  it("should let you delete a database", () => {
```
[the remark this answers is not in the document]
Comment (nemanjaglumac) [reply frontend/test/metabase/scenarios/admin/databases/list.cy.spec.js:40]: There seems to be one already:
https://github.com/metabase/metabase/blob/75d23718428d8efb6bb8a64bb677cca1d511a7de/frontend/test/metabase/scenarios/admin/databases/edit.cy.spec.js#L240

**Review by daltojohnso — APPROVED** [review]

**Pull request merged**